packages/devtools_app/lib/src/shared/table.dart

```diff
@@ -318,9 +325,16 @@ class Selection<T> {
     this.nodeIndex,
     this.nodeIndexCalculator,
     this.scrollIntoView = false,
-  });
+  }) : assert(nodeIndex != null || nodeIndexCalculator != null);
```

Comment:
this assert will still be true even if both nodeIndex and nodeIndexCalculator are specified. Change to `assert(nodeIndex == null || nodeIndexCalculator == null)` so that it will fail when both `nodeIndex` is non-null and `nodeIndexCalculator` is non-null, but will not fail when both are null.